frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -495,28 +517,56 @@ function MainMenu(props: Props): ReactElement {
     }
   }
 
+  const { hostIsOwner } = props
+
+  const showDevelopmentMenu = (() => {
+    if (props.toolbarMode == Config.ToolbarMode.DEVELOPER) {
+      return true
+    } else if (
+      props.toolbarMode == Config.ToolbarMode.VIEWER ||
+      props.toolbarMode == Config.ToolbarMode.MINIMAL
+    ) {
+      return false
+    } else {
+      return hostIsOwner || isLocalhost()
+    }
+  })()
+
   const devMenuItems: any[] = []
-  let devLastMenuItem = null
-  for (const devMenuItem of preferredDevMenuOrder) {
-    if (devMenuItem) {
-      if (devMenuItem !== coreDevMenuItems.DIVIDER) {
-        if (devLastMenuItem === coreDevMenuItems.DIVIDER) {
-          devMenuItems.push({ ...devMenuItem, hasDividerAbove: true })
-        } else {
-          devMenuItems.push(devMenuItem)
+
+  if (showDevelopmentMenu) {
+    const preferredDevMenuOrder: any[] = [
+      coreDevMenuItems.developerOptions,
+      coreDevMenuItems.clearCache,
+      showDeploy && coreDevMenuItems.deployApp,
+    ]
+
+    let devLastMenuItem = null
+
+    for (const devMenuItem of preferredDevMenuOrder) {
+      if (devMenuItem) {
+        if (devMenuItem !== coreDevMenuItems.DIVIDER) {
+          if (devLastMenuItem === coreDevMenuItems.DIVIDER) {
+            devMenuItems.push({ ...devMenuItem, hasDividerAbove: true })
+          } else {
+            devMenuItems.push(devMenuItem)
+          }
         }
+
+        devLastMenuItem = devMenuItem
       }
+    }
 
-      devLastMenuItem = devMenuItem
+    if (devLastMenuItem != null) {
+      devLastMenuItem.styleProps = lastDevMenuItemStyleProps
```

Comment:
Again, this looks big and complex enough to get extracted to a separate function.